File: app/src/main/java/com/vaibhav/moviedetails/network/ApiInterface.java
package com.vaibhav.moviedetails.network;

import com.vaibhav.moviedetails.data.Movie;
import com.vaibhav.moviedetails.data.MovieResponse;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Query;

/**
 * @author Vaibhav Bhandula on 09/10/18.
 */
public interface ApiInterface {

    @GET("/")
    Observable<MovieResponse> getMovieList(@Query("page") int mPage, @Query("s") String mQuery);

    @GET("/")
    Observable<Movie> getMovieDetails(@Query("i") String imdbId);
}

File: app/src/main/java/com/vaibhav/moviedetails/network/ApiInterface.kt
package com.vaibhav.moviedetails.network

import com.vaibhav.moviedetails.data.Movie
import com.vaibhav.moviedetails.data.MoviesResponse
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Query


/**
 * @author Vaibhav Bhandula on 08/04/19.
 */
interface ApiInterface {

    @GET("/")
    fun getMovieList(@Query("page") mPage: Int, @Query("s") mQuery: String): Observable<MoviesResponse>

    @GET("/")
    fun getMovieDetails(@Query("i") imdbId: String): Observable<Movie>
}
